x-pack/plugins/ingest_pipelines/public/application/components/pipeline_processors_editor/context/processors_context.tsx

```diff
@@ -149,6 +156,8 @@ export const PipelineProcessorsContextProvider: FunctionComponent<Props> = ({
           });
           break;
         case 'managingProcessor':
+          setUnsavedProcessorFormData(processorTypeAndOptions);
```

Comment:
Great suggestion. I went ahead and implemented this, but only update the ref on submit rather than the `form.subscribe` callback, as the form data ends up getting reset on tab change here.